processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -343,18 +345,33 @@ private void initPredicate()
             return;
           }
 
-          final boolean hasLowerLongBound;
-          final boolean hasUpperLongBound;
-          final long lowerLongBound;
-          final long upperLongBound;
+          boolean hasLowerLongBound;
+          boolean hasUpperLongBound;
+          long lowerLongBound;
+          long upperLongBound;
           boolean matchesNothing = false;
 
           if (hasLowerBound()) {
             final Long lowerLong = GuavaUtils.tryParseLong(lower);
             if (lowerLong == null) {
-              // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
-              hasLowerLongBound = false;
-              lowerLongBound = 0L;
+              BigDecimal lowerBigDecimal = getBigDecimalLowerBoundFromFloatString(lower);
+              if (lowerBigDecimal == null) {
+                // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
+                hasLowerLongBound = false;
+                lowerLongBound = 0L;
+              } else {
+                try {
+                  lowerLongBound = lowerBigDecimal.longValueExact();
+                  hasLowerLongBound = true;
+                } catch (ArithmeticException ae) { // the BigDecimal can't be contained in a long
```

Comment:
ArithmeticException indicates that the BigDecimal either had a fractional portion, or is too big for a long

The first case won't happen here because the BigDecimal gets floor/ceil applied prior to this, so this is strictly for handling the large numbers. 

The lower bound is still checked in the sense that a lower bound < Long.MIN_VALUE is treated as "match everythinig on lower bound", while a lower bound > Long.MAX_VALUE is treated as "match nothing"

I think this should remain, I don't see a strong reason for not allowing it